File: tests/test_command_session_list.py
# pyright: standard

from pathlib import Path

from typer.testing import CliRunner

from aico.main import app

runner = CliRunner()


def test_session_list_shows_active_and_all_views(tmp_path: Path) -> None:
    with runner.isolated_filesystem(temp_dir=tmp_path):
        # GIVEN a shared-history project with one default view
        result_init = runner.invoke(app, ["init"])
        assert result_init.exit_code == 0

        # WHEN I list sessions
        result_list_1 = runner.invoke(app, ["session-list"])
        # THEN it shows the default 'main' view as active
        assert result_list_1.exit_code == 0
        assert "Available sessions:" in result_list_1.stdout
        assert "  - main (active)" in result_list_1.stdout

        # WHEN I create a new session and list again
        result_new = runner.invoke(app, ["session-new", "dev"])
        assert result_new.exit_code == 0

        result_list_2 = runner.invoke(app, ["session-list"])
        # THEN both sessions are listed and 'dev' is active
        assert result_list_2.exit_code == 0
        assert "  - main" in result_list_2.stdout
        assert "  - dev (active)" in result_list_2.stdout

File: tests/test_command_session_list.rs
use assert_cmd::cargo_bin_cmd;
use tempfile::tempdir;

#[test]
fn test_session_list_shows_active_and_all_views() {
    let temp = tempdir().unwrap();
    let root = temp.path();

    // GIVEN a shared-history project with one default view
    cargo_bin_cmd!("aico")
        .current_dir(root)
        .arg("init")
        .assert()
        .success();

    // WHEN I list sessions
    let output1 = cargo_bin_cmd!("aico")
        .current_dir(root)
        .arg("session-list")
        .assert()
        .success()
        .get_output()
        .stdout
        .clone();
    let stdout1 = String::from_utf8(output1).unwrap();

    // THEN it shows the default 'main' view as active
    assert!(stdout1.contains("Available sessions:"));
    assert!(stdout1.contains("  - main (active)"));

    // WHEN I create a new session and list again
    cargo_bin_cmd!("aico")
        .current_dir(root)
        .args(["session-new", "dev"])
        .assert()
        .success();

    let output2 = cargo_bin_cmd!("aico")
        .current_dir(root)
        .arg("session-list")
        .assert()
        .success()
        .get_output()
        .stdout
        .clone();
    let stdout2 = String::from_utf8(output2).unwrap();

    // THEN both sessions are listed and 'dev' is active
    assert!(stdout2.contains("  - main"));
    assert!(stdout2.contains("  - dev (active)"));
}
